pkg/gardenlet/controller/controllerinstallation/controllerinstallation/reconciler_test.go

```diff
@@ -0,0 +1,134 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package controllerinstallation_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/ptr"
+
+	. "github.com/gardener/gardener/pkg/gardenlet/controller/controllerinstallation/controllerinstallation"
+)
+
+var _ = Describe("Reconciler", func() {
+	var (
+		reconciler *Reconciler
+	)
+
+	BeforeEach(func() {
+		reconciler = &Reconciler{}
+	})
+
+	Describe("#BootstrapControlPlaneNodeFunc", func() {
+		var (
+			pod         *corev1.Pod
+			deployment  *appsv1.Deployment
+			statefulSet *appsv1.StatefulSet
+			daemonSet   *appsv1.DaemonSet
+			job         *batchv1.Job
+			cronJob     *batchv1.CronJob
+		)
+
+		BeforeEach(func() {
+			pod = &corev1.Pod{}
+			deployment = &appsv1.Deployment{}
+			statefulSet = &appsv1.StatefulSet{}
+			daemonSet = &appsv1.DaemonSet{}
+			job = &batchv1.Job{}
+			cronJob = &batchv1.CronJob{}
+		})
+
+		It("should not change objects if not bootstrapping control plane node", func() {
+			reconciler.BootstrapControlPlaneNode = false
+
+			p := pod.DeepCopy()
+			Expect(reconciler.BootstrapControlPlaneNodeFunc(p)).To(Succeed())
```

Comment:
Nit: can you inline `pod.DeepCopy` in the function call?

Reply:
Maybe I missed it, but how can I inline the call if I want to compare the result of the call after the function call? Just looking at these two lines inlining is the way to go, but there is a line below that uses both `p` and `pod`. With inlining, one of them does not exist anymore.

Reply:
Stupid me. You're right, of course.